lib/samlp.js

```diff
@@ -211,19 +211,20 @@ module.exports.auth = function(options) {
 
   return function (req, res, next) {
     getSamlRequest(req.query.SAMLRequest || req.body.SAMLRequest, function(err, samlRequestDom) {
+      var audience = options.audience;
       if (samlRequestDom) {
         var issuer = xpath.select("//*[local-name(.)='Issuer' and namespace-uri(.)='urn:oasis:names:tc:SAML:2.0:assertion']/text()", samlRequestDom);
-        if (issuer && issuer.length > 0) options.audience = options.audience || issuer[0].textContent;
+        if (issuer && issuer.length > 0) audience = issuer[0].textContent || audience;
```

Comment:
Ok I understand, my fix isn't correct, but bug stay present if you don't set options.audience
You store it in options.audience which is global 
https://github.com/auth0/node-samlp/blob/master/lib/samlp.js#L230
Next time and locked options.audience is setted by previous value
https://github.com/auth0/node-samlp/blob/master/lib/samlp.js#L230
So issuer[0].textContent is never called again
https://github.com/auth0/node-samlp/blob/master/lib/samlp.js#L230